dataland-csvconverter/src/main/kotlin/org/dataland/csvconverter/csv/EuTaxonomyForNonFinancialsCsvParser.kt

```diff
@@ -0,0 +1,60 @@
+package org.dataland.csvconverter.csv
+
+import org.dataland.csvconverter.csv.CompanyInformationCsvParser.Companion.companyInformationColumnMapping
+import org.dataland.csvconverter.csv.CsvUtils.NOT_AVAILABLE_STRING
+import org.dataland.csvconverter.csv.CsvUtils.getCsvValue
+import org.dataland.csvconverter.csv.CsvUtils.getNumericCsvValue
+import org.dataland.csvconverter.csv.EuTaxonomyUtils.getAttestation
+import org.dataland.csvconverter.csv.EuTaxonomyUtils.getReportingObligation
+import org.dataland.datalandbackend.model.EuTaxonomyDataForNonFinancials
+import org.dataland.datalandbackend.model.EuTaxonomyDetailsPerCashFlowType
+
+/**
+ * This class contains the parsing logic for the eu-taxonomy-non-financials framework
+ */
+class EuTaxonomyForNonFinancialsCsvParser : CsvFrameworkParser<EuTaxonomyDataForNonFinancials> {
+
+    private val columnMappingEuTaxonomyForNonFinancials = mapOf(
+        "totalRevenue" to "Total Revenue EURmm",
+        "totalCapex" to "Total CapEx EURmm",
+        "totalOpex" to "Total OpEx EURmm",
+        "eligibleRevenue" to "Eligible Revenue",
+        "eligibleCapex" to "Eligible CapEx",
+        "eligibleOpex" to "Eligible OpEx",
+        "alignedRevenue" to "Aligned Revenue",
+        "alignedCapex" to "Aligned CapEx",
+        "alignedOpex" to "Aligned OpEx",
+        "companyType" to "IS/FS"
+    )
+
+    override fun validateLine(row: Map<String, String>): Boolean {
+        // Skip all lines with financial companies or without market cap
+        return companyInformationColumnMapping.getCsvValue("companyType", row) !in listOf("FS", NOT_AVAILABLE_STRING) &&
```

Comment:
Vielleicht sollte man hier eher direkt auf "IS" gehen anstatt alles zu nehmen was nicht na und nicht FS ist. Erscheint mir logischer und auch konsistenter mit den financials. Dann kann man sich den Kommentar auch sparen das ist dann ja selbserklärend.

Reply:
Jup. Done